Eisek Part 1

DSGMods opened the pull request:
- What is the purpose of the pull request?
Adds a new NPC merchant who also serves to gate early access to the dragon TF behind a quest

- Give a brief description of what you changed or added.
New NPC
New PlaceType
New job perk + special perk for new NPC
5 new clothing items for new  NPC outfit
1 new clothing item for silly mode
1 new weapon for silly mode
4 new Items for new NPC to sell
3 new quests (1 exclusive to silly mode)
4 new quest items (1 exclusive to silly mode)
4 new (one time) encounters (2 exclusive to silly mode)
1 new combat encounter exclusive to silly mode
Some modifications to Elis market dialogue and WorldType to support new PlaceType
Some modifications to Monica, Hale/Imsu, and Fae's dialogue to support 2nd new quest
Dragonfruit moved from Ralph's inventory to new NPC
New ItemTag for new NPC
Some minor changes to dragon lore

- Are any new graphical assets required?
All required assets included and optimized in SVGO

- Has this change been tested? If so, mention the version number that the test was based on.
Tested on 0.4.6.4

- So we have a better idea of who you are, what is your Discord Handle?
dsg#4931

DSGMods committed:
Initial Eisek work

DSGMods committed:
Fixed hair issue
Added parser support with unique speech color
Added initial daily schedule

DSGMods committed:
Initial quest and dialogue work

DSGMods committed:
More dialogue work

DSGMods committed:
Merge pull request #28 from Innoxia/dev

Merge 4.0.5 changes

DSGMods committed:
Merge pull request #31 from Innoxia/dev

Merge 4.0.6

DSGMods committed:
Merge pull request #35 from Innoxia/dev

Merge 4.0.8 changes

DSGMods committed:
Merge pull request #38 from Innoxia/dev

Merge 4.0.9 changes

DSGMods committed:
0.4.10 compatibility patch

DSGMods committed:
0.4.10 compatibility patch again

DSGMods committed:
0.4.10 compatibility patch again again

DSGMods committed:
Merge pull request #41 from Innoxia/dev

Merge 0.4.10 changes

DSGMods committed:
0.4.10 compatibility patch again again again

DSGMods committed:
0.4.11 compatibility patch

DSGMods committed:
Merge pull request #45 from Innoxia/dev

Merge 0.4.1.1 changes

DSGMods committed:
0.4.11 compatibility patch again

DSGMods committed:
Merge pull request #47 from Innoxia/dev

Merge 0.4.1.2 changes

DSGMods committed:
Merge pull request #49 from Innoxia/dev

Merge 0.4.1.2 changes

DSGMods committed:
Merge pull request #51 from Innoxia/dev

Merge 0.4.1.5

DSGMods committed:
0.4.2 compatibility patch

DSGMods committed:
Merge pull request #54 from Innoxia/dev

Merge 0.4.2 changes

DSGMods committed:
0.4.2 compatibility patch again

DSGMods committed:
Merge pull request #56 from Innoxia/dev

Merge 0.4.2.3 changes

DSGMods committed:
Merge pull request #58 from Innoxia/dev

Merge 0.4.2.3 final changes

DSGMods committed:
Merge pull request #60 from Innoxia/dev

Merge 0.4.2.5

DSGMods committed:
Merge pull request #62 from Innoxia/dev

Merge 0.4.2.9 changes

DSGMods committed:
Merge pull request #64 from Innoxia/dev

Merge 0.4.3 changes

DSGMods committed:
Merge pull request #67 from Innoxia/dev

Merge v0.4.3 changes

DSGMods committed:
Merge pull request #69 from Innoxia/dev

Merge v0.4.3.3 changes

DSGMods committed:
Merge pull request #71 from Innoxia/dev

Merge 0.4.3.5 changes

DSGMods committed:
Merge pull request #73 from Innoxia/dev

Merge v0.4.3.8 changes

DSGMods committed:
0.4.3.9 compatibility patch

DSGMods committed:
Merge pull request #77 from Innoxia/dev

Merge 0.4.3.9 changes

DSGMods committed:
0.4.3.9 compatibility patch

DSGMods committed:
Fix missing import

DSGMods committed:
Merge pull request #79 from Innoxia/dev

0.4.4 pre-merge

DSGMods committed:
Merge pull request #80 from Innoxia/dev

0.4.4 changes

DSGMods committed:
Merge pull request #84 from Innoxia/dev

Merge 0.4.4.1 PRs

DSGMods committed:
0.4.4.2 compat patch

DSGMods committed:
Merge pull request #87 from Innoxia/dev

Merge 0.4.4.2 changes

DSGMods committed:
Merge pull request #89 from Innoxia/dev

Merge v0.4.4.5 preview

DSGMods committed:
Merge pull request #92 from Innoxia/dev

Merge 0.4.5 changes

DSGMods committed:
Add Eisek's Ring item

DSGMods committed:
0.4.5.2 compat patch

DSGMods committed:
0.4.5.2 compat patch again

DSGMods committed:
Merge pull request #96 from Innoxia/dev

Merge 0.4.5.2 changes

DSGMods committed:
0.4.5.2 compat patch again again

DSGMods committed:
Finish up outfit work, update some icons

DSGMods committed:
Add items for trade

DSGMods committed:
Patch for 0.4.6 compatibility

DSGMods committed:
Merge pull request #102 from Innoxia/dev

Merge 0.4.6 changes

DSGMods committed:
Patch for 0.4.6 compatibility again

DSGMods committed:
Finish up item work
Clothe Eisek and fill merchant inventory

DSGMods committed:
Add PlaceType icons
Initial PlaceType work
Pre-compat patch commit

DSGMods committed:
0.3.4.6 compat patch

DSGMods committed:
0.3.4.6 compat patch again

DSGMods committed:
Merge pull request #104 from Innoxia/dev

Merge 0.4.6.3 changes

DSGMods committed:
0.3.4.6 compat patch again again

DSGMods committed:
Finish Eisek's initial encounter, stall interaction, and first quest
Standardize file locations
Edit some lore
Add poster graphics to first quest
Fixed all initial naming bugs

DSGMods committed:
Fix many bugs and NPE
Polish some scenes
Known issue: PlaceType name for Eisek's stall is not parsing

AceXP commented on res/dialogue/dsg/characters/eisek/stall.xml:
```diff
@@ -0,0 +1,394 @@
+				<responseTabIndex><![CDATA[0]]></responseTabIndex>
+				<index><![CDATA[5]]></index>
+				<responseTitle><![CDATA[Mob]]></responseTitle>
+				<responseTooltip><![CDATA[Probably best to not bring up this topic back up without some results to show.]]></responseTooltip>
```
Should be: Probably best to not bring this topic back up without some results to show.

AceXP commented on res/dialogue/dsg/characters/eisek/stall.xml:
```diff
@@ -0,0 +1,394 @@
+				<responseTabIndex><![CDATA[0]]></responseTabIndex>
+				<index><![CDATA[1]]></index>
+				<responseTitle><![CDATA[Focus on mob]]></responseTitle>
+				<responseTooltip><![CDATA[Come up with a solution with Eisek's problem.]]></responseTooltip>
```
I think it should be: Come up with a solution to Eisek's problem.

AceXP commented on res/dialogue/dsg/encounters/fields/elis/eisek_mob_hideout.xml:
```diff
@@ -0,0 +1,414 @@
+		
+		<tooltip/>				
+		<preParsingEffects><![CDATA[
+			[#game.addNPC('fields.ElisAlleywayAttacker', 'muscle')]
```
I think it's always a good idea to prefix new NPC parser tags to prevent possible conflicts.

So could be DSGmuscle in this case.

AceXP commented on res/items/dsg/eisek/apples.xml:
```diff
@@ -0,0 +1,57 @@
+
+	<useDescriptions>
+		<selfUse><![CDATA[
+			[npc.Name] [npc.verb(find)] the tangerine surprisingly easy to peel and [npc.verb(start)] eating it wedge by wedge. It doesn't take long to consume the whole thing.
```
Should refer to apples here, not tangerines. They don't come in cartons as well.

AceXP commented on res/items/dsg/eisek/tangerines.xml:
```diff
@@ -0,0 +1,70 @@
+			[npc.Name] [npc.verb(find)] the tangerine surprisingly easy to peel and [npc.verb(start)] eating it wedge by wedge. It doesn't take long to consume the whole thing.
+		]]></selfUse>
+		<otherUse><![CDATA[
+			[npc.Name] [npc.verb(find)] the tangerine surprisingly easy to peel and [npc.verb(start)] feeding it to [npc2.name] wedge by wedge. It doesn't take long to empty the carton after that...
```
No carton for tangerines.

AceXP commented on res/txt/dsg/encounters/fields/elis/eisek_mob_hideout.xml:
```diff
@@ -0,0 +1,213 @@
+		#ELSEIF(pc.hasSpell(SPELL_POISON_VAPOURS))
+			You summon a few wisps of a poisonous vapour and swirl them around the room.
+		#ELSEIF(pc.hasSpell(SPELL_ICE_SHARD))
+			You cast a shard of ice right that flies right between the donkey-boy's legs and freezes patch of floor beneath his chair.
```
Should be like this I think: You cast a shard of ice that flies right between the donkey-boy's legs and freezes a patch of floor beneath his chair.

DSGMods committed:
Fix some typos found by @AceXP
Also change "muscle" enemy tag to avoid potential conflict

DSGMods committed:
Fix some (more) typos found by @AceXP

DSGMods committed:
Missed some NPC tags

DSGMods committed:
Fix PlaceType parsing
Fix incorrect values used for debugging

DSGMods committed:
Add variant of PlaceType description for when player has not yet met Eisek and he is not present
Add low affection variant for talk_business
Add lots of variants for personality traits and observant perk

DSGMods committed:
Add @AceXP backwards compatibility for saves that already visited the market

DSGMods committed:
Fix farmer perk icon and description as well as standardize icon filename
Adjust ring description

DSGMods committed:
Change sell multiplier value for Eisek
Add items for second quest
Change dragonfruit author tag to match other produce items
Clean up old dialogue and text files
Add dialogue for second quest to Elis vendors
Edit existing second quest dialogue
Edit second quest stage descriptions slightly

DSGMods committed:
Initial second quest dialogue work
Fix missing conditionals and parsing errors found by @AceXP

DSGMods committed:
Fix second quest node tree
Fix many, many dialogue bugs in second quest

DSGMods committed:
Add variations to PlaceType dialogue to reflect choices made during 1st quest
Add items for 3rd quest for silly mode

DSGMods committed:
Small dialogue edits
Initial 3rd quest work

DSGMods committed:
0.4.6.7 compat patch

DSGMods committed:
Merge pull request #108 from Innoxia/dev

Merge 0.4.6.7 changes

DSGMods committed:
0.4.7 compat patch

DSGMods committed:
Merge pull request #110 from Innoxia/dev

Merge 0.4.7 changes

DSGMods committed:
Added checks for players that are already dragons

DSGMods committed:
Actually pay the player on initial encounter like the dialogue says
Change sellModifier to 1 as stall quest reward

DSGMods committed:
Rename half-demon dragons

DSGMods committed:
Merge pull request #112 from Innoxia/dev

Merge 0.4.7.5 changes

DSGMods committed:
Update all food items to have the 'FOOD' tag

DSGMods committed:
Rewrite Eisek sillymode encounter slightly
Add status effect tag support to food items

DSGMods commented:
Converting back to draft since Silly Mode content is now beyond the point of no return.

DSGMods committed:
New quest item

DSGMods committed:
Added sillymode encounter and questline
Added sillymode dungeon map and basic enemy
Added new item for player to loot at the end
Moved stall place dialogue to match txt directory
Various small edits

DSGMods committed:
Moar dungeon work

DSGMods committed:
Even MOAR dungeon work
Added combat encounter
Also shuffled files around to make IDs less dumb

DSGMods committed:
Merge branch 'Innoxia:dev' into eisek_part_one_compat_patch

DSGMods committed:
Finished dungeon work
Added "boss" encounter
Fixed some incorrect text directories in dialogue

DSGMods committed:
Cleanup placeTypes
Add new unique icons

DSGMods committed:
Lots of bugs and formatting fixes.

DSGMods committed:
Change the debug value in the sillymode encounter back to normal

Innoxia commented:
Thank you so much for this! 🐉
